specification/sections/03-evaluation-context.md

```diff
@@ -42,15 +42,39 @@ see: [structure](../types.md#structure), [datetime](../types.md#datetime)
 
 The key uniquely identifies a field in the `evaluation context` and it should be unique across all types to avoid any collision when marshalling the `evaluation context` by the provider.
 
-### 3.2 Merging Context
+### 3.2 Context levels and merging
 
-#### Requirement 3.2.1
+#### Condition 3.2.1
+
+> The implementation uses the static-context paradigm.
+
+see: [static-context paradigm](../glossary.md#static-context-paradigm)
```

Comment:
Not sure where the correct part in the spec is to ask about this, but what about providing a callback mechanism at the Variable level for variable value updates when the context is changed?

From this [blog post](https://openfeature.dev/blog/catering-to-the-client-side/) it seems like the intention was to use events like this: 
```
client.addHandler(ProviderEvents.FlagValuesChanged, () => {
  // this would trigger a re-render
  setUseFormalSalutation(client.getBooleanValue('use-formal-salutation', false));
});
```

I would recommend that a flag / variable level callback handler would be more useful for users and make integrations into React Hooks and other similar implementations easier.

```typescript
openFeatureClient.getBooleanValue('boolean-flag', false)
    .onUpdate((value: boolean, details: EvaluationDetails<boolean>) => {

    })
```

Reply:
I'm not certain if this is part of the implementation (going to test it out soon), but does the OpenFeature SDK have a mechanism of being told when the underlying provider has updated its feature cache? if so then this could be handled within the OpenFeature SDKs. But you are likely correct that this is likely too specific of an implementation detail for this spec right now. However, as we get into implementing more reactive / hook-based client-side SDKs this almost becomes necessary to propagate flag updates.

Reply:
@jonathannorris have you had a chance to confirm if `PROVIDER_CONFIGURATION_CHANGED` covers the case you mentioned?